parcels/fieldset.py

```diff
@@ -178,15 +178,26 @@ def from_netcdf(cls, filenames, variables, dimensions, indices=None,
         fields = {}
         for var, name in variables.items():
             # Resolve all matching paths for the current variable
-            paths = filenames[var] if type(filenames) is dict else filenames
-            if not isinstance(paths, list):
-                paths = sorted(glob(str(paths)))
-            if len(paths) == 0:
-                notfound_paths = filenames[var] if type(filenames) is dict else filenames
-                raise IOError("FieldSet files not found: %s" % str(notfound_paths))
-            for fp in paths:
-                if not path.exists(fp):
-                    raise IOError("FieldSet file not found: %s" % str(fp))
+            paths = filenames[var] if type(filenames) is dict and var in filenames else filenames
+            if type(paths) is not dict:
+                if not isinstance(paths, list):
+                    paths = sorted(glob(str(paths)))
+                if len(paths) == 0:
+                    notfound_paths = filenames[var] if type(filenames) is dict and var in filenames else filenames
+                    raise IOError("FieldSet files not found: %s" % str(notfound_paths))
+                for fp in paths:
+                    if not path.exists(fp):
+                        raise IOError("FieldSet file not found: %s" % str(fp))
+            else:
+                for dim, p in paths.items():
+                    if not isinstance(p, list):
+                        paths[dim] = sorted(glob(str(p)))
+                    if len(paths[dim]) == 0:
+                        notfound_paths = filenames[var] if type(filenames) is dict and var in filenames else filenames
+                        raise IOError("FieldSet files not found: %s" % str(notfound_paths))
+                    for fp in paths[dim]:
+                        if not path.exists(fp):
+                            raise IOError("FieldSet file not found: %s" % str(fp))
```

Comment:
Here, we have two copies of almost the same code. Create a function instead, for conciseness an ease of maintenance?